tests/DocumentationExamplesTest.php

```diff
@@ -1585,17 +1570,19 @@ public function testSnapshotQueries(): void
             $this->markTestSkipped('Snapshot read concern is only supported with replicasets');
         }
 
+        $this->dropCollection('pets', 'cats');
```

Comment:
Since `dropCollection()` now returns a Collection object, you could remove the three sequential `dropCollection()` calls here and have them replace the original `selectCollection()` calls below:

```diff
-$catsCollection = $client->selectCollection('pets', 'cats');
-$catsCollection->drop();
+$catsCollection = $this->dropCollection('pets', 'cats');
 $catsCollection->insertMany([
```

Doing so will mean that `$client = static::createTestClient();` is no longer needed so early in the function, so I'd suggest moving that further down as well:

```diff
+$client = static::createTestClient();

 ob_start();

 // Start Snapshot Query Example 1
```

Also, the three `Collection::drop()` calls outside of start/end example blocks can be removed, since you're handling that in the new `tearDown()` logic.

```diff
 $this->assertSame(3, $adoptablePetsCount);

-$catsCollection->drop();
-$dogsCollection->drop();
```

```diff
 $this->assertSame(1, $totalDailySales);

-$salesCollection->drop();
```